File: android/app/src/main/java/com/cartoncloud/application/WeatherApplication.java
package com.cartoncloud.application;

import android.app.Application;

import com.cartoncloud.BuildConfig;
import com.cartoncloud.react.ViewPackager;
import com.facebook.react.ReactApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;

import java.util.ArrayList;
import java.util.List;

/**
 * Application class to handle the ViewModule
 */
public class WeatherApplication extends Application implements ReactApplication {

    private ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {

        @Override
        public boolean getUseDeveloperSupport() {
            return BuildConfig.DEBUG;
        }

        @Override
        protected List<ReactPackage> getPackages() {
            List<ReactPackage> packages = new ArrayList<>();
            packages.add(new MainReactPackage());
            packages.add(new ViewPackager());
            return packages;
        }
    };

    @Override
    public ReactNativeHost getReactNativeHost() {
        return mReactNativeHost;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        //SoLoader.init(this, /* native exopackage */ false)

    }
}

File: android/app/src/main/java/com/cartoncloud/application/WeatherApplication.kt
package com.cartoncloud.application

import android.app.Application
import com.cartoncloud.react.ViewPackager
import com.facebook.react.BuildConfig
import com.facebook.react.ReactApplication
import com.facebook.react.ReactNativeHost
import com.facebook.react.shell.MainReactPackage
import com.facebook.soloader.SoLoader

/**
 * Application class to handle the ViewModule
 */
class WeatherApplication : Application(), ReactApplication {

    private val mReactNativeHost = object : ReactNativeHost(this) {
        override fun getUseDeveloperSupport() = BuildConfig.DEBUG

        override fun getPackages() = listOf(
                MainReactPackage(),
                ViewPackager()
        )
    }

    override fun getReactNativeHost() = mReactNativeHost

    override fun onCreate() {
        super.onCreate()
        SoLoader.init(this, /* native exopackage */ false)
    }
}